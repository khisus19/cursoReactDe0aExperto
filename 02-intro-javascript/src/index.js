

const api_key = "9hNhYfR5jnRYPkOi33k6MxzUwtPNqJFt";

const peticion = fetch(`api.giphy.com/v1/gifs/random?api_key=${ api_key }`);

peticion.then( resp => resp.json() )
  .then( ({ data }) => {
    console.log(data.images.original.url);
  })
  .catch( console.warn );