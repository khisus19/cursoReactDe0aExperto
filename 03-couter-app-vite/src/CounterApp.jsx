import { useState } from 'react';
import PropTypes from "prop-types";


export const CounterApp = ({ value }) => {
  
  const [ counter, setCounter ] = useState(value);

  const handleAdd = () => setCounter( counter + 1);
  const handleSub = () => setCounter( counter - 1);
  const handleReset = () => setCounter( value );

  return (
    <>
      <h1>Counter App</h1>
      <p>{ counter }</p>
      <button onClick={ handleAdd } >+1</button>
      <button onClick={ handleSub } >-1</button>
      <button onClick={ handleReset } >Reset</button>
    </>
  )
}

CounterApp.propTypes = {
  value: PropTypes.number,
}

CounterApp.defaultProps = {
  name: "Jesús Álvarez",
  subtitle: "No hay subtitulo",
  // title: "No hay titulo",
}