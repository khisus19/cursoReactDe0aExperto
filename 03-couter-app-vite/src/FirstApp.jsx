
export const FirstApp = () => {
  const funcionPrueba = ( a, b = 2) => {
    return a + b
  } 


  return <h1>{ funcionPrueba( 5, 5) }</h1>
}