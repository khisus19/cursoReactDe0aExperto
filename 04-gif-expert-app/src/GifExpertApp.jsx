import { useState } from "react";
import { AddCategory } from "./components/AddCategory";

export const GifExpertApp = () => {
  
  const [categories, setCategories] = useState([ "One punch", "Dragon ball" ]);

  const onAddCategory = () => {
    setCategories([...categories, "Inuyasha"])
  }

  return (
    <>
      {/* titulo */}
      <h1>GifExpertApp</h1>

      {/* Input */}
      <AddCategory />

      {/* Listado de gif */}
      <button onClick={ onAddCategory }>Agregar Categoria</button>
      <ol>
        { categories.map(category => {
          return <li key={category}>{ category }</li>
        }) }
      </ol>
        {/* Gif Item*/}



    </>
  )
}
