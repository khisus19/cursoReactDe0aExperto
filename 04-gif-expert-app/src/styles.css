* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
  font-family: 'Manrope', sans-serif;
  color: var(--clr-light);
}
:root {
  --clr-main: #00d9f6;
  --clr-dark: #0e121b;
  --clr-grey: #6c7589;
  --clr-light: #fff;
  --ff-secondary: 'Source Sans Pro', sans-serif;
}

/* NAVBAR */
nav {
  width: 100%;
  padding: 20px;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-between;
  font-weight: 600;
}
#logo img {
  width: 250px;
}
@media (min-width: 600px) {
  nav {
    display: flex;
    flex-direction: row;
  }
}
.nav-link-ul {
  width: 100%;
  max-width: 350px;
  list-style: none;
  font-size: 1rem;
  font-weight: 600;
  display: flex;
  justify-content: space-evenly;
}
li {
  margin: 5px;
}
a, a:visited, a:focus {
  text-decoration: none;
  color: var(--clr-main);
}
a:hover {
  color: var(--clr-grey);
}

/* Body */
body {
  background-color: var(--clr-dark);
}
main {
  padding: 25px;
  display: flex;
  flex-direction: column;
  align-items: center;
}

form {
  margin: 15px auto;
  max-width: 600px;
  width: 90%;
}
input {
  background-color: white;
  border-radius: 5px;
  border: 1px solid rgb(97, 32, 158);
  color: black;
  font-size: 1.2rem;
  outline: none;
  padding: 10px 15px;
  width: 100%;
}

h2 {
  font-size: 1.5rem;
}

h3 {
  font-size: 3rem;
  margin: 5px auto;
  text-align: center;
}

.card-grid {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
}

.card {
  padding: 10px;
  align-content: center;
  align-items: center;
  background-color: white;
  border-radius: 10px;
  border: 2px double var(--clr-grey);
  display: flex;
  flex-direction: column;
  height: 0%;  
  justify-content: center;
  margin-bottom: 20px;
  margin-right: 20px;
  overflow: hidden;
}

.card p {
  background-color: white;
  color: var(--clr-dark);
  flex: 1;
  font-size: 1.5rem;
  font-weight: 600;
  margin-top: 5px;
  padding: 10px 20px;
  text-align: center;
}

.card img {
  width: 100%;
}