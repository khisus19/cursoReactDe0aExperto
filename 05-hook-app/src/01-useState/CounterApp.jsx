

export const CounterApp = () => {

  const [counter, setCounter] = useState(10);


  return (
    <>
      <h1>Counter:</h1>

      <hr />

      <button onClick={ () => setCounter( counter + 1 ) }>+1</button>

    </>
  )
}
