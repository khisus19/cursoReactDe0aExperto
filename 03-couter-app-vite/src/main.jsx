import React from 'react';
import ReactDOM from 'react-dom/client';
import { FirstApp } from "./FirstApp";
import { App } from './HelloWorldApp'


ReactDOM.createRoot( document.getElementById("root")).render(
  <React.StrictMode>
    <App />
    <FirstApp />
    
  </React.StrictMode>
)