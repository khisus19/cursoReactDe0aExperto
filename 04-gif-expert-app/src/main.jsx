import React from 'react';
import ReactDOM from 'react-dom/client';
import { GifExpertApp } from "./GifExpertApp";
import "./styles.css"
import { NavBar } from "./components/NavBar";


ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <NavBar />
    <main>
      <GifExpertApp />
    </main>
  </React.StrictMode>,
);
