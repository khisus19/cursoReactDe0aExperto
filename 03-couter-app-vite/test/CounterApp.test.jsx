import { fireEvent, render, screen } from "@testing-library/react";
import { CounterApp } from "../src/CounterApp";


describe('Prueba de CounterApp', () => { 
  
  test('Debe hacer match con el snapshot', () => { 
   
    const { container } = render( <CounterApp value={ 100 } /> );
    expect( container ).toMatchSnapshot();

  })
  
  test('Debe tener el valor de 100', () => { 
   
    render( <CounterApp value={ 100 } /> );
    expect( screen.getByText(100) ).toBeTruthy();

  })

  test('Debe tener el valor de 100 dentro de una etiqueta <p>', () => { 
   
    render( <CounterApp value={ 100 } /> );
    // console.log(screen);
    expect( screen.getByRole("heading", {level: 3}).innerHTML ).toContain("100");

  })

  test('debe de incrementar con el boton +1', () => { 
    
    render( <CounterApp value={ 100 } />);
    fireEvent.click( screen.getByText("+1"));
    expect( screen.getByText("101") ).toBeTruthy();

  })

  test('debe de decrementar con el boton -1', () => { 
    
    render( <CounterApp value={ 100 } />);
    fireEvent.click( screen.getByText("-1"));
    expect( screen.getByText("99") ).toBeTruthy();

  })
  
  test('debe de restaurar el value con el boton reset', () => { 
    
    render( <CounterApp value={ 100 } />);
    fireEvent.click( screen.getByText("-1"));
    fireEvent.click( screen.getByText("-1"));
    
    fireEvent.click( screen.getByRole("button", { name: "btn-reset" }) );
    expect( screen.getByText("100") ).toBeTruthy();

  })

})