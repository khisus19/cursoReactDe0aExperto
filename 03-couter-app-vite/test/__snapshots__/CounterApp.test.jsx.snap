// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba de CounterApp Debe hacer match con el snapshot 1`] = `
<div>
  <h1>
    Counter App
  </h1>
  <h3>
    100
  </h3>
  <button>
    +1
  </button>
  <button>
    -1
  </button>
  <button
    aria-label="btn-reset"
  >
    Reset
  </button>
</div>
`;
